ee/tabby-schema/src/schema/integration.rs

```diff
@@ -93,7 +97,12 @@ pub trait IntegrationService: Send + Sync {
         api_base: Option<String>,
     ) -> Result<ID>;
 
-    async fn delete_integration(&self, id: ID, kind: IntegrationKind) -> Result<()>;
+    async fn delete_integration(
+        &self,
+        id: ID,
+        kind: IntegrationKind,
+        third_party_repository_service: Arc<dyn ThirdPartyRepositoryService>,
```

Comment:
service should be part of `self`, instead of pass in as argument

Reply:
That would create a circular dependency between `ThirdPartyRepositoryService` and `IntegrationService`, since the repository service already contains the integration service. This could be done with a `Weak<Mutex<Option<Arc<dyn ThirdPartyIntegrationService>>>>`, but would we rather avoid that?

Reply:
Tabled